voip_oca/models/mail_activity_mixin.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2025 Dixmit
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ResConfigSettings(models.AbstractModel):
+    _inherit = "mail.activity.mixin"
+
+    activity_main_partner_id = fields.Many2one(
+        "res.partner",
+        string="Main Contact",
+        compute="_compute_activity_main_partner_id",
+    )
+
+    def _compute_activity_main_partner_id(self):
+        for record in self:
+            record.activity_main_partner_id = record.get_activity_main_partner_id()
+
+    def get_activity_main_partner_id(self):
+        if self._name == "res.partner":
+            return self
+        if "partner_id" in self._fields:
+            return self.partner_id
+        return self.env["res.partner"]
```

Comment:
No need of 2 methods. For inheritance, just override the compute:

```suggestion
    def _compute_activity_main_partner_id(self):
        self.activity_main_partner_id = False
        if "partner_id" in self._fields:
            for record in self:
                record.activity_main_partner_id = record.partner_id
```

and then on res.partner:

```
    def _compute_activity_main_partner_id(self):
        for record in self:
            record.activity_main_partner_id = record
```

Reply:
It is done this way in order to leave a hook for other modules.

Reply:
As said, the hook can be done in the compute method in the mixin or in the specific model. No need of extra method.

Reply:
Yes, but it is easier to extend it this way. Also, there are examples of this behavior inside OCA and Odoo. I would prefer to keep it as it is

Reply:
I don't think so, as the stacktrace is longer, and the code is less readable being split and this method is only called in one place, which is the clear indicator that shouldn't be extracted.

Reply:
Well, it was done this way to have it done as it is right now (I forgot to split it correctly). With this,  we can change it by thread model. It is longer, yes, but it is cleaner too IMO. Also, we can find several examples in Odoo Core:

https://github.com/odoo/odoo/blob/c315f5af7bbe47d6fa73d0dd0ba31ab6f93cfd03/addons/mail/models/mail_thread.py#L429
https://github.com/odoo/odoo/blob/c315f5af7bbe47d6fa73d0dd0ba31ab6f93cfd03/addons/hr_recruitment/models/hr_job.py#L339

Reply:
Well, that examples are not exactly the same and Odoo core can't be sometimes the best example, but this is not so important to block the PR.